File: backend/schemas.py
# Definição dos schemas Pydantic (modelos de dados da API)
from pydantic import BaseModel
from typing import Optional, List
import datetime

# --- Schemas de Tarefas ---

class TaskBase(BaseModel):
    title: str
    description: Optional[str] = None

class TaskCreate(TaskBase):
    pass

class Task(TaskBase):
    id: int
    completed: bool # <-- CAMPO ADICIONADO DE VOLTA
    created_at: datetime.datetime
    owner_id: int

    class Config:
        from_attributes = True

# --- Schemas de Usuários (sem alterações) ---

class UserBase(BaseModel):
    email: str

class UserCreate(UserBase):
    password: str

class User(UserBase):
    id: int
    is_active: bool
    tasks: List[Task] = []

    class Config:
        from_attributes = True

# --- Schemas de Token (sem alterações) ---

class Token(BaseModel):
    access_token: str
    token_type: str

class TokenData(BaseModel):
    email: Optional[str] = None

File: frontend/src/lib/schemas.ts
/**
 * Define a estrutura (schema) dos dados que vêm da API.
 * Isso ajuda o TypeScript a entender os dados e evitar erros.
 */

export interface Task {
  id: number;
  title: string;
  description: string | null;
  completed: boolean;
  created_at: string; // A API retorna a data como uma string no formato ISO
  owner_id: number;
}

export interface User {
  id: number;
  email: string;
  is_active: boolean;
}
